src/database/arangodb/aql-generator.ts

```diff
@@ -1319,39 +1319,49 @@ function getQuantifierFilterUsingArrayComparisonOperator(
 
     const valueFrag = processNode(conditionNode.rhs, context);
 
-    let fieldValueFrag: AQLFragment;
+    let fieldValueFrag: () => AQLFragment;
```

Comment:
Why did you change this field to a function that resolves to an `AQLFragment` ?

Reply:
the value of `isSafeList` can now change *after* the conditional here. I could untangle the logic so it first fully calculates `isSafeList`, then calculates the `fieldValueFrag`, and then uses it. However, the logic of whether or not `isSafeList` can safely be set to `false` depends on how the `fieldValueFrag` is used, so I think it's clearer to set that flag to false where we use the `fieldValueFlag`.

I just changed it a bit again, renamed it to `fieldValueFragFn`, added a parameter `useSafeListWrapper` to make it more explicit, and added comments

Do you think it's easier to understand now?